core/src/main/java/org/mule/runtime/core/internal/event/AbstractEventContext.java

```diff
@@ -37,130 +37,147 @@
  */
 abstract class AbstractEventContext implements BaseEventContext {
 
+  private static final int STATE_READY = 0;
```

Comment:
no enum then?

Reply:
no. using an enum with ordinal is less clear and there is no real advantage.